### DISCUSSION TITLE
Docs for authentication with separate server

### DISCUSSION THREAD

**Discussion by tar-aldev** [vercel/next.js#22013] [Ideas]
**Describe the feature you'd like to request**

I am quite new to next.js so sorry if this question is not a good fit.
I see there are some examples for authentication, but all of them are using next js api folder.
What if I have separate server which is used for authentication? How would I handle session based authentication (if it is possible at all)?
I could get session id via `context.req.headers.cookie`, but I have no luck to somehow use it further, to attach it to my requests.

**Describe the solution you'd like**

Some examples (or explanation) on how to setup authentication with separate api server.

**Describe alternatives you've considered**

I looked into with-next-auth example, but it seems I cannot use it for this use case.

**Comment by tar-aldev** [comment]
If it helps I am using express-session on backend and session based authentication.
I used to make it work by the following code
```
const { data } = await axios.get<GetAuthenticatedUserResponse>(
      'http://localhost:5000/api/users/me',
      {
        withCredentials: true,
        headers: {
          Cookie: context.req.headers.cookie
        }
      }
    );
```
But I am still not sure about 
1. Will it always work even if I go to this page via next.js Link component?
2. Is it safe to attach cookies like this? As far as I understand this request will happen on server, not client, so it should be fine.
Any help is much appreciated

**Reply by sanjampreet-singh** [reply]
As far as your code is visible there should not be any problem, this is how the cookies are passed by browser via headers. But since you are using Next,js server you need to do this work manually. I have failed to use express session with Next.js, if you can guide me with few steps or any links will be appreciated.

**Reply by Raghuboi** [reply]
I've been trying to implement it myself, I can generate the cookies and store the sessions, but can't seem to figure out how to authenticate the session using cookies. My apologies, I am very new to session based authentication, if any of you has figured it out since then I would appreciate some guidance as well.
